pkg/cmd/opts/version.go

```diff
@@ -3,16 +3,96 @@ package opts
 import (
 	"strings"
 
-	"github.com/jenkins-x/jx/pkg/versionstream/versionstreamrepo"
-
-	"github.com/jenkins-x/jx/pkg/versionstream"
-
+	"github.com/jenkins-x/jx/pkg/config"
+	"github.com/jenkins-x/jx/pkg/envctx"
+	"github.com/jenkins-x/jx/pkg/kube"
 	"github.com/jenkins-x/jx/pkg/log"
 	"github.com/jenkins-x/jx/pkg/table"
 	"github.com/jenkins-x/jx/pkg/util"
 	"github.com/jenkins-x/jx/pkg/version"
+	"github.com/jenkins-x/jx/pkg/versionstream"
+	"github.com/jenkins-x/jx/pkg/versionstream/versionstreamrepo"
 )
 
+// EnvironmentContext gets or creates a team context with the common values for working with requirements, team settings
+// and version resolvers.
+//
+// if preferRequirementsFile is true we look for the local `jx-requirements.yml` file first and then fall back to the
+// development environment settings; otherwise we try to use the requirements from the environment if present
+func (o *CommonOptions) EnvironmentContext(dir string, preferRequirementsFile bool) (*envctx.EnvironmentContext, error) {
+	if o.envctx != nil {
+		return o.envctx, nil
+	}
+	var err error
+	tc := &envctx.EnvironmentContext{}
+	tc.GitOps, tc.DevEnv = o.GetDevEnv()
+	if tc.DevEnv == nil {
+		tc.DevEnv = kube.CreateDefaultDevEnvironment("jx")
+	}
+	teamSettings := tc.TeamSettings()
+
+	// lets default to local file system for the requirements as we are often invoked before we've created the cluster
+	exists := false
+	if preferRequirementsFile {
+		fileName := ""
+		tc.Requirements, fileName, err = config.LoadRequirementsConfig(dir)
+		if err != nil {
+			return tc, err
+		}
+		if fileName != "" {
+			exists, _ = util.FileExists(fileName)
+		}
+	}
+	if !preferRequirementsFile || !exists {
+		// lets try the environment CRD if we have no local file
+		req, err := config.GetRequirementsConfigFromTeamSettings(teamSettings)
+		if err != nil {
+			return tc, err
+		}
+		if req != nil {
+			tc.Requirements = req
+		}
+	}
+	if err != nil {
+		return tc, err
+	}
+
+	// if we can't find a requirements then lets just create the defaults for now
+	if tc.Requirements == nil {
+		tc.Requirements, _, err = config.LoadRequirementsConfig(dir)
+		if err != nil {
+			return tc, err
+		}
+	}
+	err = o.ConfigureCommonOptions(tc.Requirements)
+	if err != nil {
+		return tc, err
+	}
+	versionStreamURL := teamSettings.VersionStreamURL
+	versionStreamRef := teamSettings.VersionStreamRef
```

Comment:
Order seems odd here - why pull from teamSettings before the newly build env context?

Reply:
let me modify that to use either the requirements file or the environment/team settings based on the new boolean which indicates the priority (based on whether its file based, during the boot pipeline, or environment based after-boot). Sound ok?

Reply:
ok we use either the local requirements or the environment team settings based on the flag: https://github.com/jenkins-x/jx/blob/00c0cee8775cc8d8ed2fab0ba158a8997330a9f2/pkg/cmd/opts/version.go#L74

Reply:
👍